src/esp/sensor/VisualSensor.h

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#pragma once
+
+#include <Corrade/Containers/Optional.h>
+
+#include "esp/core/esp.h"
+
+#include "esp/gfx/RenderCamera.h"
+#include "esp/gfx/RenderTarget.h"
+#include "esp/sensor/Sensor.h"
+
+namespace esp {
+namespace sensor {
+
+// Represents a sensor that provides data from the environment to an agent
```

Comment:
May be good to make this a bit more constrained to `provides visual data` since this is a `VisualSensor`